src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,297 @@
-        logger.debug("INTERNET is required for Firebase Analytics.");
-        return null;
-      }
+          Context context = analytics.getApplication();
 
-      Context context = analytics.getApplication();
-
-      return new FirebaseIntegration(context);
-    }
+          return new FirebaseIntegration(context, logger);
+        }
 
-    @Override public String key() {
-      return FIREBASE_ANALYTICS_KEY;
-    }
-  };
+        @Override
+        public String key() {
+          return FIREBASE_ANALYTICS_KEY;
+        }
+      };
 
-  private static final String FIREBASE_ANALYTICS_KEY = "Firebase Analytics";
+  static final Pattern ISO_DATE =
+          Pattern.compile("(^([0-9]{4})-(1[0-2]|0[1-9])-(0[1-9]|1[1-9]|2[0-9]|3[0-1]).*)", CASE_INSENSITIVE);
+  private static final String FIREBASE_ANALYTICS_KEY = "Firebase";
+  final Logger logger;
+  final FirebaseAnalytics mFirebaseAnalytics;
 
-  public FirebaseIntegration(Context context) {
+  public FirebaseIntegration(Context context, Logger logger) {
+    this.logger = logger;
     mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
   }
 
-  @Override public void onActivityStarted(Activity activity) {
-    super.onActivityStarted(activity);
-
-  }
+  @Override
+  public void onActivityResumed(Activity activity) {
+    super.onActivityResumed(activity);
 
-  @Override public void onActivityStopped(Activity activity) {
-    super.onActivityStopped(activity);
+    PackageManager packageManager = activity.getPackageManager();
+    try {
+      ActivityInfo info =
+              packageManager.getActivityInfo(activity.getComponentName(), PackageManager.GET_META_DATA);
+      CharSequence activityLabel = info.loadLabel(packageManager);
+      /** setCurrentScreen should only be called in the onResume() activity */
+      mFirebaseAnalytics.setCurrentScreen(
+              activity, activityLabel.toString(), null /* class override */);
+      logger.verbose(
+              "mFirebaseAnalytics.setCurrentScreen(%s, %s, null /* class override */);",
+              activity, activityLabel.toString());
 
+    } catch (PackageManager.NameNotFoundException e) {
+      throw new AssertionError("Activity Not Found: " + e.toString());
+    }
   }
 
-  @Override public void identify(IdentifyPayload identify) {
+  @Override
+  public void identify(IdentifyPayload identify) {
     super.identify(identify);
 
     if (!isNullOrEmpty(identify.userId())) {
       mFirebaseAnalytics.setUserId(identify.userId());
-    }
 
-    // mFirebaseAnalytics.setUserProperty(key, value);
+      Map<String, Object> traits = identify.traits();
+
+      for (Map.Entry<String, Object> entry : traits.entrySet()) {
+        String trait = entry.getKey();
+
+        trait = trait.trim().replaceAll(" ", "_");
+
+        if (trait.length() > 40) {
+          trait = trimKey(trait);
+        }
+
+        String formattedValue;
+
+        if (entry.getValue() instanceof Date) {
+
```

Comment:
There's a lot of blank lines around the braces. These are unnecessary.  

This is the style guide we should be following https://google.github.io/styleguide/javaguide.html#s4.6.1-vertical-whitespace